pyvrp/Model.py

```diff
@@ -0,0 +1,281 @@
+           can be added to the model.
+        """
+        if len(self._depots) >= 1:
+            msg = "PyVRP does not yet support multi-depot VRPs."
+            raise ValueError(msg)
+
+        depot = Depot(x, y, tw_early=tw_early, tw_late=tw_late)
+        self._depots.append(depot)
+        return depot
+
+    def add_edge(
+        self,
+        frm: Union[Client, Depot],
+        to: Union[Client, Depot],
+        distance: int,
+        duration: int = 0,
+    ) -> Edge:
+        """
+        Adds an edge :math:`(i, j)` between ``frm`` (:math:`i`) and ``to``
+        (:math:`j`). The edge can be given distance and duration attributes.
+        Distance is required, but the default duration is zero. Returns the
+        created edge.
+
+        Raises
+        ------
+        ValueError
+            When either distance or duration is a negative value.
+        """
+        if distance < 0 or duration < 0:
+            raise ValueError("Cannot have negative edge distance or duration.")
+
+        edge = Edge(frm, to, distance, duration)
+        self._edges.append(edge)
+        return edge
+
+    def add_vehicle_type(self, number: int, capacity: int) -> VehicleType:
+        """
+        Adds a vehicle type with the given number of vehicles of given capacity
+        to the model.
+
+        .. warning::
+
+           PyVRP does not yet support heterogeneous fleet VRPs. For now, only
+           one vehicle type can be added to the model.
+
+        Raises
+        ------
+        ValueError
+            When either the vehicle number or capacity is not a positive value.
+        """
+        if len(self._vehicle_types) >= 1:
+            msg = "PyVRP does not yet support heterogeneous fleet VRPs."
+            raise ValueError(msg)
+
+        if number <= 0:
+            raise ValueError("Must have positive number of vehicles.")
+
+        if capacity < 0:
+            raise ValueError("Cannot have negative vehicle capacity.")
+
+        vehicle_type = VehicleType(number, capacity)
+        self._vehicle_types.append(vehicle_type)
+        return vehicle_type
+
+    def update(self):
+        """
+        Updates the underlying ProblemData instance.
+        """
+        clients = self._depots + self._clients
+        client2idx = {id(client): idx for idx, client in enumerate(clients)}
+
+        num_vehicles = self._vehicle_types[0].number
+        vehicle_capacity = self._vehicle_types[0].capacity
+
+        # This should be a sufficiently large maximum value to make sure such
+        # edges are never traversed.
+        max_value = 10 * max(max(e.distance, e.duration) for e in self._edges)
+        if max_value > 0.1 * np.iinfo(np.int32).max:  # >10% of INT_MAX
+            msg = """
+            The maximum distance or duration value is very large. This might
+            impact numerical stability. Consider rescaling your input data.
+            """
+            warn(msg, ScalingWarning)
```

Comment:
The former probably yes, the latter.. maybe? I cannot find anything in the pybind docs about warnings, only errors. Warnings are a Python-level thing, so it might be really hard. Only adding it to read seems easier.